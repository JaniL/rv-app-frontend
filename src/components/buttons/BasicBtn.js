import React from "react";
import "./styles/Button.css";
import "./styles/BasicBtn.css";
import Loader from "./../loaders/Loader";

const SuccessBtn = ({ onClick, children, fill, hover, loader, ...props }) => {
  let className = "btn";
  fill ? (className += " basic-fill") : (className += " basic");
  hover &&
    (fill ? (className += " basic-fill-hover") : (className += " basic-hover"));

  return (
    <button {...props} onClick={onClick} className={className}>
      {!loader ? <span className="btnContent">{children}</span> : <Loader />}
    </button>
  );
};

export default SuccessBtn;
