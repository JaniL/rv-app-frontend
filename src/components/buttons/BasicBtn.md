### Without hover

```js
<div>
  <BasicBtn>Hello world</BasicBtn> <BasicBtn fill>Hello world</BasicBtn>
</div>
```

### With hover

```js
<div>
  <BasicBtn fill hover>
    Hello world
  </BasicBtn> <BasicBtn hover>Hello world</BasicBtn>
</div>
```

### With loader

```js
<div>
    <BasicBtn loader fill /> <BasicBtn loader />
</div>
```

### With click handler

```js
const clickHandler = () => {
  alert("Hello world");
};
<div>
    <BasicBtn fill onClick={clickHandler}>
    Click me for Hello World
    </BasicBtn> <BasicBtn onClick={clickHandler}>
    Click me for Hello World
    </BasicBtn>
</div>
```
