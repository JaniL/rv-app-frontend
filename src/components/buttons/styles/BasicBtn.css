/*
  Basic buttons
  */
.basic {
  color: #269bd1;
  -webkit-transition: background-color 0.1s;
  /* Safari */
  transition: background-color 0.1s; }

.basic:hover {
  color: white;
  background-color: #269bd1;
  border-color: #269bd1;
  cursor: pointer; }

.basic-fill {
  color: white;
  background-color: #269bd1;
  border-color: #269bd1;
  -webkit-transition: background-color 0.1s;
  /* Safari */
  transition: background-color 0.1s; }

.basic-fill:hover {
  color: #269bd1;
  background-color: transparent;
  border-color: #269bd1;
  cursor: pointer; }
