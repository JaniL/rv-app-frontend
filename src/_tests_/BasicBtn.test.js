import React from 'react';
import ReactDOM from 'react-dom';
import BasicBtn from '../components/buttons/BasicBtn';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BasicBtn />, div);
    const cmpnt = renderer.create(<BasicBtn />).toJSON();
    expect(cmpnt).toMatchSnapshot();
});
