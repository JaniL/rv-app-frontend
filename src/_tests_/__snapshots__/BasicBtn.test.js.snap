// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  className="btn basic"
  onClick={null}
>
  <span
    className="btnContent"
  />
</button>
`;
